experiments/integrated/fluxnet/plot_utils.jl

```diff
@@ -0,0 +1,77 @@
+"""Plotting utilities for the integrated fluxnet site experiments"""
+
+using Interpolations
+using Plots
+using StatsBase
+
+S_PER_DAY = 86400 # Number of seconds in a day
+
+"""This function uses interpolation to convert a time series of data at 
+any regular time interval to half hourly data. It takes in the data, the 
+vector giving the time stamps of the data in seconds, and the number of days 
+in the data series. It returns a vector of the data interpolated to half
+hourly intervals."""
+function interp_hh(data::Vector, timeseries::Vector, num_days::Int64)
+    # Rescale the data to a half hourly interval by linear interpolation
+    interp = LinearInterpolation(timeseries, data)
+    half_hourly = 0.5*3600:0.5*3600:24 * num_days * 3600
+    hh_data = [interp[i] for i in half_hourly]
+    return hh_data
+end
+
+"""This function takes in a vector of half-hourly data and the number of days 
+in the data series and returns the vector giving the diurnal average of 
+the data over the time period. (For data which is not half-hourly, use 
+interp_hh first.)"""
+function diurnal_avg(hh_series::Vector, num_days::Int64)
+    # Reshape the data into a matrix with each column representing a day
+    daily_data = reshape(hh_series, 48, num_days)
+
+    # Average each row over all days to get diurnal average
+    hh_avgs = mean(daily_data, dims=2)
+    return hh_avgs
+end
+
+"""Given a time series of data, the corresponding time axis, and the number 
+of days in the data, linearly interpolates the data to a half hourly time 
+scale, and returns a vector giving the diurnal average of the data over the time
+period."""
+function diurnal_avg_hh(data::Vector, timeseries::Vector, num_days::Int64)
```

Comment:
if i understand correctly, this is a helper function which uses the two previously defined functions